### ISSUE TITLE
Timeouts Not Working

### ISSUE THREAD

**Issue by dirges** [xetorthio/jedis#1125]
I'm evaluating Jedis and can't seem to get any of the timeout parameters to work. Here's the test code I'm using. It's pointing at a Redis instance that has been purposely stopped.

From looking at the code, this constructor should use the timeout parameter for both the connect and read timeouts but even if I use a different constructor I get the same result. That is, the call to "get" never returns.

```java
Jedis jedis = new Jedis("localhost", 6379, 5000);
String value = jedis.get("test"); // this blocks indefinitely
```

I am on Mac OSX 10.11 and I'm using Jedis 2.7.3.

**Comment by marcosnils** [comment]
@dirges how / when did you stop the redis instance to generate a timeout?

**Comment by dirges** [comment]
The redis instance is a local one that I'm running in debug mode. I'm just manually stopping the process before I run the test case above.

**Comment by marcosnils** [comment]
@dirges which Jedis version are you running?

**Comment by marcosnils** [comment]
@dirges something is weird here. If you say you're stopping your redis instance before running the above code Jedis / Java throws a `java.net.ConnectException: Connection refused` exception when trying to connect to an ip/port that's not listening. 

If you try to telnet localhost 6379 after stopping your redis instance. What do you get?

**Comment by dirges** [comment]
@marcosnils I feel like a dope but it appears the problem was on my end. I took a closer look at my process list and there was a second instance of redis running that seemed to be in some kind of unknown state. What's strange is that when I ran redis on a console in debug mode it never complained about being unable to bind to the default port and was definitely responding to my requests, yet, when I stopped it (and the old instance was still apparently running), Jedis behaved like I described. When I finally killed the old process and repeated my tests by starting/stopping the debug instance, everything worked as expected. I appreciate your help. Thanks.
